arithmetization/src/main/java/net/consensys/linea/zktracer/module/romlex/RomLex.java

```diff
@@ -241,15 +241,22 @@ public void callRomLex(final MessageFrame frame) {
 
   @Override
   public void resolvePostExecution(
-      Hub hub, MessageFrame frame, Operation.OperationResult operationResult) {
-    Preconditions.checkArgument(hub.opCode().isCreate());
-    final int depNumber = hub.transients().conflation().deploymentInfo().number(this.address);
+      Hub hub, MessageFrame frame, Operation.OperationResult operationResult) {}
+
+  @Override
+  public void resolveUponImmediateContextEntry(Hub hub) {
+    checkArgument(hub.messageFrame().getType() == MessageFrame.Type.CONTRACT_CREATION);
+
+    final int deploymentNumber = hub.deploymentNumberOf(address);
+    final boolean deploymentStatus = hub.deploymentStatusOf(address);
     final ContractMetadata contractMetadata =
-        ContractMetadata.underDeployment(this.address, depNumber);
+        ContractMetadata.underDeployment(address, deploymentNumber);
+
+    checkArgument(deploymentStatus, "After a CREATE the deployment status should be true");
 
-    final RomChunk chunk = new RomChunk(contractMetadata, true, false, this.byteCode);
-    this.chunks.add(chunk);
-    this.createDefers.trigger(contractMetadata);
+    final RomChunk chunk = new RomChunk(contractMetadata, true, false, byteCode);
+    chunks.add(chunk);
+    createDefers.trigger(contractMetadata);
```

Comment:
CREATE's are deferred to `ContextEntry`

I kept 
```java
createDefers.trigger(contractMetadata);
```
as I didn't look into what it does.